File: 2021-SOPM/E2/E02-01/app/src/main/java/com/example/e02_01/WelcomeActivity.java
package com.example.e02_01;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class WelcomeActivity extends AppCompatActivity {

    final String FULL_NAME_KEY = "FULL_NAME_KEY";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);

        Intent mainActivityIntent = getIntent();

        if (mainActivityIntent != null) {
            String name = mainActivityIntent.getStringExtra(FULL_NAME_KEY);
            TextView welcomeMessage = findViewById(R.id.welcome_text);

            String welcomeTextToDisplay = getString(R.string.welcome_text, name);
            welcomeMessage.setText(welcomeTextToDisplay);
        }
    }
}

File: 2021-SOPM/E2/E02-01/app/src/main/java/com/example/e02_01/WelcomeActivity.kt
package com.example.e02_01

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.e02_01.R
import android.content.Intent
import android.widget.TextView

class WelcomeActivity : AppCompatActivity() {
    private val FULL_NAME_KEY = "FULL_NAME_KEY"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_welcome)
        val mainActivityIntent = intent
        if (mainActivityIntent != null) {
            val name = mainActivityIntent.getStringExtra(FULL_NAME_KEY)
            val welcomeMessage = findViewById<TextView>(R.id.welcome_text)
            val welcomeTextToDisplay = getString(R.string.welcome_text, name)
            welcomeMessage.text = welcomeTextToDisplay
        }
    }
}